src/libraries/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -49,6 +50,7 @@ private static string MarshalToString(IntPtr arg, string argName)
         /// <param name="delegateTypeNative">Assembly qualified delegate type name</param>
         /// <param name="reserved">Extensibility parameter (currently unused)</param>
         /// <param name="functionHandle">Pointer where to store the function pointer result</param>
+        [RequiresDynamicCode(NativeAOTIncompatibleWarningMessage)]
```

Comment:
Why do we need the `RequiresDynamicCode` warning here? I would expect that the trim compatibility warning is going to be enough.

Reply:
There is a leaf level [API in Marshal](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs#L1113) class that are annotated with `RequiresDynamicCode ` that are used by this API and I couldn't see a way to use the generic API